### ISSUE TITLE
False-positive "Match must be exhaustive" when matching pair of references

### ISSUE THREAD

**Issue by dima74** [intellij-rust/intellij-rust#5901]
## Environment

* **IntelliJ Rust plugin version:** 0.3.129.3274-202-nightly
* **Rust toolchain version:** 1.47.0-nightly (6c8927b0c 2020-07-26) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2020.2 (CL-202.6397.106)
* **Operating system:** Linux 5.7.12-arch1-1

## Problem description
![image](https://user-images.githubusercontent.com/6505554/89736737-f342c900-da84-11ea-84af-812878a78c3e.png)

Warning disappears if type of `a` is changed from `&E` to `E`

## Steps to reproduce
```rust
enum E { A, B }
use E::*;
fn foo(a: &E, b: &E) {
    match (a, b) {
        (A, A) => {}
        (_, B) => {}
        (B, A) => {}
    }
}

```

**Comment by johnw42** [comment]
Here's a slightly more complicated example derived from the code where I encountered this issue:

    enum E {
        A,
        B,
        C,
    }
    use E::*;
    fn foo(x: &E, y: &E) {
        match (x, y) {
            (A, A) => {}
            (B, B) => {}
            (C, C) => {}
            (B, _) => {}
            (_, B) => {}
            (C, _) => {}
            (_, C) => {}
        }
    }

**Comment by dima74** [comment]
Looks like it is fixed by #5720
